src/core/net/dnssd_server.cpp

```diff
@@ -836,6 +841,38 @@ bool Server::CanAnswerQuery(const Server::QueryTransaction &aQuery, const char *
     return (sdType == kDnsQueryResolveHost) && (strcmp(name, aHostFullName) == 0);
 }
 
+const char *Server::GetDomainSplit(const char *aName, DnsQueryType aQueryType)
+{
+    uint8_t     skipDot = 0;
+    const char *split   = aName;
+
+    switch (aQueryType)
+    {
+    case kDnsQueryBrowse:
+        skipDot = 2;
+        break;
+    case kDnsQueryResolve:
+        skipDot = 3;
```

Comment:
Instance name itself may contain dots. So, the number of dots to skip is not fixed for instance resolving.